persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/BsonRecordField.scala

```diff
@@ -27,59 +27,70 @@ import net.liftweb.record._
 
 /** Field that contains an entire record represented as an inline object value. Inspired by JSONSubRecordField */
-class BsonRecordField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
-                        (rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
-  extends Field[SubRecordType, OwnerType]
-  with MandatoryTypedField[SubRecordType]
-{
-  def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: SubRecordType)
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, value.meta)
-      set(value)
-  }
+abstract class BsonRecordTypedField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
+(override val owner: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
+  extends Field[SubRecordType, OwnerType] {
 
   def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: Box[SubRecordType])
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, valueMeta)
-      setBox(value)
+    (implicit subRecordType: Manifest[SubRecordType]) = {
+    this(rec, valueMeta)
+    setBox(value)
   }
 
-  def owner = rec
-  def asJs = asJValue match {
+  override def asJs = asJValue match {
     case JNothing => JsNull
     case jv => new JsExp {
       lazy val toJsCmd = compactRender(jv)
     }
   }
-  def toForm: Box[NodeSeq] = Empty
-  def defaultValue = valueMeta.createRecord
 
-  def setFromString(s: String): Box[SubRecordType] = valueMeta.fromJsonString(s)
+  override def toForm: Box[NodeSeq] = Empty
 
-  def setFromAny(in: Any): Box[SubRecordType] = in match {
+  override def setFromString(s: String): Box[SubRecordType] = valueMeta.fromJsonString(s)
+
+  override def setFromAny(in: Any): Box[SubRecordType] = in match {
     case dbo: DBObject => setBox(Full(valueMeta.fromDBObject(dbo)))
     case dbo: Document => setBox(Full(valueMeta.fromDocument(dbo)))
     case _ => genericSetFromAny(in)
   }
 
-  def asJValue: JValue = valueBox.map(_.asJValue) openOr (JNothing: JValue)
-  def setFromJValue(jvalue: JValue): Box[SubRecordType] = jvalue match {
+  override def asJValue: JValue = valueBox.map(_.asJValue) openOr (JNothing: JValue)
+
+  override def setFromJValue(jvalue: JValue): Box[SubRecordType] = jvalue match {
     case JNothing|JNull if optional_? => setBox(Empty)
     case _ => setBox(valueMeta.fromJValue(jvalue))
   }
 }
 
+class BsonRecordField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
+(@deprecatedName('rec) owner: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
+  extends BsonRecordTypedField(owner, valueMeta) with MandatoryTypedField[SubRecordType] {
+
+  def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: SubRecordType)(implicit subRecordType: Manifest[SubRecordType]) = {
```

Comment:
I think we also need the `rec`->`owner` change here (with the `deprecatedName` annotation)